radar_control/radar_control.py

```diff
@@ -228,315 +285,188 @@ def radar():
     For every pulse, samples and other control information are sent to the n200_driver.
     For every pulse sequence, processing information is sent to the signal processing block.
     After every integration time (AveragingPeriod), the experiment block is given the opportunity
-    to change the control program (if it sends a new one, runradar will halt the old one and begin with
-    the new ControlProg.
+    to change the control program (if it sends a new one, runradar will halt the old one and begin 
+    with the new experiment).
     """
 
-    # Setup socket to send pulse samples over.
-    txsocket=setup_driver_socket()
     # Initialize driverpacket.
-    driverpacket=driverpacket_pb2.DriverPacket()
+    driverpacket = driverpacket_pb2.DriverPacket()
+
+    # Get config options.
+    options = ExperimentOptions()
+
+    context = zmq.Context()  # single context for all sockets involved in this process.
+    # Setup sockets.
+    # Socket to send pulse samples over.
+    try:
+        tx_socket = setup_socket(context, options.radar_control_to_driver_address)
+        rx_socket = setup_socket(context, options.radar_control_to_rx_dsp_address)
+        rx_ack_socket = setup_socket(context, options.rx_dsp_to_radar_control_ack_address)
+        rx_time_socket = setup_socket(context, options.rx_dsp_to_radar_control_timing_address)
+        experiment_socket = setup_socket(context, options.experiment_handler_to_radar_control_address)
+    except zmq.ZMQBaseError as e:
+        errmsg = "ZMQ ERROR Setting up sockets"
+        raise [ExperimentException(errmsg), e]
+
 
-    procsocket=setup_sigproc_params_socket()
-    sigprocpacket=sigprocpacket_pb2.SigProcPacket()
-    proctimesocket=setup_sigproc_timing_ack_socket()
-    proccpsocket=setup_sigproc_cpack_socket()
+    poller = zmq.Poller()
```

Comment:
We should add a little bit more description in the poller name, what is it a poller for?
Same with poller2

Example: pller -> poller_for_tx_complete
poller2 -> poller_for_completed_sequence